plugins/services/src/js/components/SDKPlanTable.tsx

```diff
@@ -0,0 +1,217 @@
+
+const getStatusTooltip = (data: ServicePlanElement): React.ReactNode | null => {
+  const { status } = data;
+
+  switch (status) {
+    case "ERROR":
+      return <Trans render="span">Execution experienced an error.</Trans>;
+    case "WAITING":
+      return <Trans render="span">Execution has been interrupted.</Trans>;
+    case "PENDING":
+      return (
+        <Trans render="span">Execution is waiting for suitable offers.</Trans>
+      );
+    case "PREPARED":
+      if (data.type === "step") {
+        return (
+          <Trans render="span">
+            The step has been evaluated, and any Tasks relevant to it have been
+            killed if necessary.
+          </Trans>
+        );
+      }
+      return (
+        <Trans render="span">
+          The phase has been evaluated, and any Tasks relevant to it have been
+          killed if necessary.
+        </Trans>
+      );
+    case "STARTING":
+      return (
+        <Trans render="span">
+          Execution has performed and is waiting to determine the success of
+          those operations.
+        </Trans>
+      );
+    case "STARTED":
+      return (
+        <Trans render="span">
+          Execution has performed and has received feedback, but not all success
+          requirements (e.g. readiness checks) have been satisfied.
+        </Trans>
+      );
+    case "IN_PROGRESS":
+      if (data.type === "phase" && data.steps) {
+        const { totalSteps, completedSteps } = data.steps.reduce(
+          (result, step) => {
+            result.totalSteps++;
+            if (step.status === "COMPLETE") {
+              result.completedSteps++;
+            }
+            return result;
+          },
+          {
+            totalSteps: 0,
+            completedSteps: 0
+          }
+        );
+        return (
+          <Trans render="span">
+            {completedSteps} out of {totalSteps} steps completed.
+          </Trans>
+        );
+      }
+      // Steps should never return "IN_PROGRESS" status, only plan & status
+      return null;
+    case "COMPLETE":
+      return <Trans render="span">Execution has completed.</Trans>;
+    default:
+      return null;
+  }
+};
+
+export const getStatusIcon = (status: ServicePlanStatus): React.ReactNode => {
+  switch (status) {
+    case "ERROR":
+      return <Icon id="circle-close" family="system" size="mini" color="red" />;
+    case "WAITING":
+      return <Icon id="yield" family="system" size="mini" color="yellow" />;
+    case "PENDING":
+      return <Icon id="yield" family="system" size="mini" color="yellow" />;
+    case "PREPARED":
+      return <Icon id="spinner" family="system" size="mini" />;
+    case "STARTING":
+      return <Icon id="spinner" family="system" size="mini" />;
+    case "STARTED":
+      return <Icon id="spinner" family="system" size="mini" />;
+    case "IN_PROGRESS":
+      return <Icon id="spinner" family="system" size="mini" />;
+    case "COMPLETE":
```

Comment:
I wonder if it would be useful to have Status ("COMPLETE", "STARTING", etc) & Type ("phase", "step", etc) enums? I've seen these strings used in a couple places in this file as well as `formatServicePlanStatus` function